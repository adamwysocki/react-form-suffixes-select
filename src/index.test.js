// index.test.js
import React from "react";
import SuffixesSelect from "./index";
import renderer from "react-test-renderer";
import Enzyme, { shallow, render, mount } from "enzyme";
import sinon, { stub } from "sinon";
import Adapter from "enzyme-adapter-react-16";

// React 16 Enzyme Adapter
Enzyme.configure({ adapter: new Adapter() });

it("should render", () => {
  const onSuffixSelect = stub();
  const component = renderer.create(<SuffixesSelect onChange={onSuffixSelect} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

it("should render without a default option if default option is turned off", () => {
  const onSuffixSelect = stub();
  const wrapper = mount(<SuffixesSelect onChange={onSuffixSelect} hasDefaultOption={false} />);
  const select = wrapper.find("select");
  expect(select.contains(<option value="null">Suffix</option>)).toEqual(false);
});

it("should render with the default option text specified", () => {
  const onSuffixSelect = stub();
  const wrapper = mount(<SuffixesSelect onChange={onSuffixSelect} defaultOptionText={"Suffix ..."} />);
  const select = wrapper.find("select");
  expect(select.contains(<option value="null">Suffix ...</option>)).toEqual(true);
});

it("should call onChange when select option is changed", () => {
  const onChange = sinon.spy();
  const event = { currentTarget: { value: 1 } };
  const wrapper = shallow(<SuffixesSelect onChange={onChange} />);
  wrapper.find("#suffix").simulate("change", event);
  expect(onChange.called).toEqual(true);
});

it("should call not call onChange when default option is selected", () => {
  const onChange = sinon.spy();
  const event = { currentTarget: { value: "null" } };
  const wrapper = shallow(<SuffixesSelect onChange={onChange} />);
  wrapper.find("#suffix").simulate("change", event);
  expect(onChange.called).toEqual(false);
});
