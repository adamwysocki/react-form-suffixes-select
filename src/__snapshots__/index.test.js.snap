// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="sc-bdVaJa dpXQOA"
>
  <select
    className=""
    id="suffix"
    name="suffix"
    onChange={[Function]}
    style={Object {}}
  >
    <option
      value="null"
    >
      Suffix
    </option>
    <option
      value={0}
    >
      II
    </option>
    <option
      value={1}
    >
      III
    </option>
    <option
      value={2}
    >
      IV
    </option>
    <option
      value={3}
    >
      CPA
    </option>
    <option
      value={4}
    >
      DDS
    </option>
    <option
      value={5}
    >
      Esq
    </option>
    <option
      value={6}
    >
      JD
    </option>
    <option
      value={7}
    >
      Jr
    </option>
    <option
      value={8}
    >
      LLD
    </option>
    <option
      value={9}
    >
      MD
    </option>
    <option
      value={10}
    >
      PhD
    </option>
    <option
      value={11}
    >
      Ret
    </option>
    <option
      value={12}
    >
      RN
    </option>
    <option
      value={13}
    >
      Sr
    </option>
    <option
      value={14}
    >
      DO
    </option>
  </select>
</div>
`;
